core/util/src/main/java/org/eclipse/rdf4j/common/iteration/Iterations.java

```diff
@@ -25,32 +26,37 @@
 
 	/**
 	 * Get a List containing all elements obtained from the specified Iteration.
-	 * 
+	 *
 	 * @param iter the Iteration to get the elements from
 	 * @return a List containing all elements obtained from the specified Iteration.
 	 */
 	public static <E, X extends Exception> List<E> asList(Iteration<? extends E, X> iter) throws X {
-		List<E> result = new ArrayList<>();
-		addAll(iter, result);
-		return result;
+		try (Stream<? extends E> stream = Iterations.stream(iter)) {
+
+			// stream.collect is slightly slower than addAll for lists
+			// return stream.collect(Collectors.toList());
+
+			List<E> list = new ArrayList<>();
+			return addAll(iter, list);
+		}
 	}
```

Comment:
Shouldn't this be reverted back to the original? It looks as if you're converting to a stream but not actaully using that stream.

Reply:
Good catch